plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -2783,10 +2783,17 @@ private void setStatus(String status, DBPMessageType messageType)
 
     private int closeExtraResultTabs(@Nullable QueryProcessor queryProcessor, boolean confirmClose, boolean keepFirstTab) {
         List<CTabItem> tabsToClose = new ArrayList<>();
+        QueryResultsContainer resultsContainer = null;
+        QueryProcessor processor = null;
         for (CTabItem item : resultTabs.getItems()) {
-            if (item.getData() instanceof QueryResultsContainer && item.getShowClose() && !isPinned(item)) {
-                QueryResultsContainer resultsProvider = (QueryResultsContainer)item.getData();
-                if (queryProcessor != null && queryProcessor != resultsProvider.queryProcessor) {
+            if (item.getData() instanceof QueryResultsContainer) {
+                resultsContainer = (QueryResultsContainer) item.getData();
+                processor = resultsContainer.queryProcessor;
```

Comment:
Therefore, this can be simplified:
```suggestion
                processor = ((QueryResultsContainer) item.getData()).queryProcessor;
```